src/back/TicketManager/src/Core/TicketManager.Core.Domain/Events/Sector.cs

```diff
@@ -1,6 +1,33 @@
+using TicketManager.Core.Domain.Common;
+
 namespace TicketManager.Core.Domain.Events;
 
-public record Sector(Guid EventId, string Name, int PriceInSmallestUnit, int NumberOfColumns, int NumberOfRows)
+public class Sector: 
+    IAggregateRoot<Guid>, IOptimisticConcurrent
 {
-    public (Guid EventId, string Name) Id => (EventId, Name);
+    public Guid Id { get; private init; }
+    public Guid EventId { get; private init; }
+    public string Name { get; private set; } = null!;
+    public int PriceInSmallestUnit { get; private set; }
+    public int NumberOfColumns { get; private set; }
+    public int NumberOfRows { get; private set; }
+    public int NumberOfSeats => NumberOfRows * NumberOfColumns;
+    private List<SeatReservation> seatReservations = new();
+    public IReadOnlyCollection<SeatReservation> SeatReservations => seatReservations.AsReadOnly();
```

Comment:
Why do you need `AsReadOnly`?

Reply:
We can change that if it breaks something